### PR TITLE
Get ciphertext column as a string

### PR DESCRIPTION
**Pull request by longjy**
Encrypt the field, and take the value of string type directly. For example, after digital encryption, the database can be saved as varchar, and the corresponding field of bean can still be numbe

### CODE REVIEW

**Comment by terrymanu** [comment]
I have a little confuse, why force get value from ResultSet by String type?
I can't get the point, what is the issue in original code?

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java [hunk 1]
```diff
@@ -44,15 +44,15 @@
...
-        if (null == value || !queryWithCipherColumn) {
-            return value;
+        Optional<Encryptor> encryptor;
+        if (!queryWithCipherColumn || !(encryptor = metaData.findEncryptor(columnIndex)).isPresent()) {
+            return mergedResult.getValue(columnIndex, type);
         }
-        Optional<Encryptor> encryptor = metaData.findEncryptor(columnIndex);
-        return encryptor.isPresent() ? encryptor.get().decrypt(value.toString()) : value;
+        String ciphertext = (String) mergedResult.getValue(columnIndex, String.class);
+        return ciphertext == null ? null : encryptor.get().decrypt(ciphertext);
```
Comment (SteNicholas) [inline encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java]: This should be `null == ciphertext` according to code style.

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java [hunk 2]
```diff
@@ -44,15 +44,15 @@
...
-        Object value = mergedResult.getValue(columnIndex, type);
-        if (null == value || !queryWithCipherColumn) {
-            return value;
+        Optional<Encryptor> encryptor;
+        if (!queryWithCipherColumn || !(encryptor = metaData.findEncryptor(columnIndex)).isPresent()) {
+            return mergedResult.getValue(columnIndex, type);
         }
-        Optional<Encryptor> encryptor = metaData.findEncryptor(columnIndex);
-        return encryptor.isPresent() ? encryptor.get().decrypt(value.toString()) : value;
+        String ciphertext = (String) mergedResult.getValue(columnIndex, String.class);
```
Comment (SteNicholas) [inline encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java]: Does the `mergeResult` include result type except String? This transformation could cause exception.

Comment (longjy) [reply encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java]: After verification, mysql is not a varchar field type, getString will not have a type exception. And encrypted field storage is generally varchar, Encryptor class
String encrypt (Object plaintext) also returns String

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java (hunk 1 above)
Comment (longjy) [reply encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java]: Coding style I will fix

**Comment by terrymanu** [comment]
Why close this pr?

**Comment by longjy** [comment]
> Why close this pr?

Misoperation, please help me reopen PR

**Pull request closed without merging**